src/Abstractions/NexusMods.Abstractions.NexusWebApi/ILoginManager.cs

```diff
@@ -38,13 +38,13 @@ public interface ILoginManager
     /// <summary>
     /// True if the user is logged in and is a premium member
     /// </summary>
-    IObservable<bool> IsPremiumObservable => UserInfoObservable.WhereNotNull().Select(static x => x.IsPremium).DistinctUntilChanged().AsSystemObservable();
+    IObservable<bool> IsPremiumObservable => UserInfoObservable.WhereNotNull().Select(static x => x.UserRole == UserRole.Premium).DistinctUntilChanged().AsSystemObservable();
     
     /// <summary>
-    /// True if the user is a supporter member
+    /// The user's role
     /// </summary>
-    IObservable<bool> IsSupporterObservable => UserInfoObservable.WhereNotNull().Select(static x => x.IsSupporter).DistinctUntilChanged().AsSystemObservable();
-
+    IObservable<UserRole?> UserRoleObservable => UserInfoObservable.Select(static x => x?.UserRole).DistinctUntilChanged().AsSystemObservable();
```

Comment:
Observable of null values is somewhat of an anti-pattern.

```suggestion
    IObservable<UserRole> UserRoleObservable => UserInfoObservable.WhereNotNull().Select(static x => x.UserRole).DistinctUntilChanged().AsSystemObservable();
```

Reply:
Thanks for this